twake/backend/node/src/services/channels/services/channel/activities/service.ts

```diff
@@ -0,0 +1,155 @@
+import { logger } from "../../../../../core/platform/framework";
+import { localEventBus } from "../../../../../core/platform/framework/pubsub";
+import { Channel as ChannelEntity } from "../../../entities/";
+import { ResourceEventsPayload } from "../../../../types";
+import { GenericObjectType } from "./types";
+import _ from "lodash";
+
+export default class Activities {
+  version = "1";
+
+  init(): Activities {
+    const channelMemberCreatedEvent = "channel:member:created";
+    localEventBus.subscribe<ResourceEventsPayload>(channelMemberCreatedEvent, data =>
+      this.logGenericData(channelMemberCreatedEvent, {
+        type: "channel:activity:member:created",
+        actor: {
+          type: "user",
+          id: data.actor.id,
+        },
+        context: {
+          type: "add",
+          array: [
+            // data.actor.id !== data.member.user_id ? "invite" : "join"
+            { type: "member", resource: data.resourcesAfter[0] },
+          ],
+        },
+      }),
+    );
+
+    const channelMemberDeletedEvent = "channel:member:deleted";
+    localEventBus.subscribe<ResourceEventsPayload>(channelMemberDeletedEvent, data =>
+      this.logGenericData(channelMemberDeletedEvent, {
+        type: "channel:activity:member:deleted",
+        actor: {
+          type: "user",
+          id: data.actor.id,
+        },
+        context: {
+          type: "remove",
+          array: [{ type: "member", resource: data.resourcesBefore[0] }],
+        },
+      }),
+    );
+
+    const channelUpdatedEvent = "channel:updated";
+    localEventBus.subscribe<ResourceEventsPayload>(channelUpdatedEvent, data => {
+      const interestedChanges = ["channel_group", "name", "description", "visibility", "icon"];
+      const channelChanged = !_.isEqual(
+        _.pick(data.resourcesBefore[0], interestedChanges),
+        _.pick(data.resourcesAfter[0], interestedChanges),
+      );
+
+      const connectorsBefore: Partial<ChannelEntity> = _.pick(data.resourcesBefore[0], [
+        "connectors",
+      ]);
+
+      const connectorsAfter: Partial<ChannelEntity> = _.pick(data.resourcesAfter[0], [
+        "connectors",
+      ]);
+
+      const channelConnectorsChanged = !_.isEqual(connectorsBefore, connectorsAfter);
```

Comment:
Pretty sure isEqual will not work well on arrays

Reply:
Googled it, you should take a look https://stackoverflow.com/questions/29951293/using-lodash-to-compare-jagged-arrays-items-existence-without-order

Reply:
`connectorsBefore` and `connectorsAfter` are objects
![Capture d’écran de 2021-02-08 13-57-37](https://user-images.githubusercontent.com/36481167/107222916-ceb8b100-6a15-11eb-82cd-38520bb1423a.png)

Reply:
They are objects but connectors array may not be ordered the same:

![Capture d’écran 2021-02-08 à 14 40 27](https://user-images.githubusercontent.com/264403/107227473-b481d180-6a1b-11eb-8b0c-bab3620e2a82.png)

So you have to order them before compare

Reply:
Thank you, i updated this 😄 
 ``` ts
const channelConnectorsChanged = !_.isEqual(
  { connectors: sortBy(connectorsBefore.connectors) },
  { connectors: sortBy(connectorsAfter.connectors) },
);
```